src/Cli/Handler/MergeHandler.php

```diff
@@ -526,4 +524,60 @@ private function addMetadata(string $sha, string $category, array $pr, array $au
         $this->git->addNotes(json_encode($metadata, \JSON_THROW_ON_ERROR), $sha, 'pr-metadata');
         $this->git->pushToRemote(REMOTE_MAIN, 'refs/notes/pr-metadata');
     }
+
+    /** @param array<string, mixed> $pr */
+    private function handleSplitOperation(Args $args, array $pr): void
+    {
+        $ref = $pr['base']['ref'];
+
+        if ($args->getOption('no-split') || $args->getOption('no-pull') || ! $this->updateLocalBranch($ref)) {
```

Comment:
This is wrong as the `$this->updateLocalBranch($ref)` is only related to `no-pull`, but when `no-split` is given this step would be skipped.